### ISSUE TITLE
[RELEASE] New release for 2024-12-13

### ISSUE THREAD

**Issue by krisstern** [jenkinsci/bom#4092]
A new release is being scheduled.
Release manager: @krisstern

# Note
For tasks that don\'t have a specific script, i.e. tasks 2, 4, 5, 7, and 10, run `./bom-release-issue-complete-task.sh <task number>` to check the box off without having to manually edit the issue.

# Release progress
* [x] 1. makes sure the CRON expression to run the job is appropriate for the release date 
  * [line to check in the Jenkinsfile](https://github.com/jenkinsci/bom/blob/master/Jenkinsfile#L4), time is UTC.
* [x] 2. run `./bom-lock-master.sh` before the job runs
* [x] 3. wait to verify that job started at [ci.jenkins.io](https://ci.jenkins.io/job/Tools/job/bom/job/master/)			
* [x] 4. run `./bom-release-issue-job-running.sh <buildNumber>`
  * Example: `./bom-release-issue-job-running.sh 1234`
* [x] 5. wait for build to make it through the `prep` stage then (typically) take a 1.5-2 hr break
* [x] 6. (LOOP) if there are any failures, fix until everything is successful
* [ ] 7. run `./bom-run-cd-workflow.sh`
  * wait for the release process to complete
  * this takes 7-8 minutes
* [x] 8. manually edit the auto-generated release notes
  * remove `<!-- Optional: add a release summary here -->`
  * remove `<details>`
  * remove `<summary>XYZ changes</summary>`
  * remove `</details>`
* [x] 9. run `./bom-release-issue-add-release-comment.sh`
* [x] 10. run `./bom-unlock-master.sh`
* [x] 11. verify that the [branch is unlocked](https://github.com/jenkinsci/bom/settings/branch_protection_rules/6421306)
* [x] 12. run `./bom-release-issue-close.sh`

**Comment by krisstern** [comment]
[https://ci.jenkins.io/job/Tools/job/bom/job/master/3564/](https://ci.jenkins.io/job/Tools/job/bom/job/master/3564/)

**Comment by krisstern** [comment]
[https://ci.jenkins.io/job/Tools/job/bom/job/master/3566/](https://ci.jenkins.io/job/Tools/job/bom/job/master/3566/)

**Comment by gounthar** [comment]
No red for the time being... 🤞

**Comment by krisstern** [comment]
There is a red somewhere for [pct-pipeline-build-step-plugin-2.462.x / org.jenkinsci.plugins.workflow.support.steps.build.WaitForBuildStepTest.abortBuild](https://ci.jenkins.io/job/Tools/job/bom/job/master/3566/testReport/junit/org.jenkinsci.plugins.workflow.support.steps.build/WaitForBuildStepTest/pct_pipeline_build_step_plugin_2_462_x___abortBuild/), let me try again once the build has completed.

**Comment by krisstern** [comment]
[https://ci.jenkins.io/job/Tools/job/bom/job/master/3568/](https://ci.jenkins.io/job/Tools/job/bom/job/master/3568/)

**Comment by darinpope** [comment]
I ran step 7 (the CD step), but [it fails](https://github.com/jenkinsci/bom/actions/runs/12326177390):

```
================================
Release job filter details:
          VALIDATE_ONLY: false
              CI_STATUS: failed
   INTERESTING_CATEGORY: 
--------------------------------
=>       SHOULD_RELEASE: false
================================
```

even though [the job appears to have succeeded](https://ci.jenkins.io/job/Tools/job/bom/job/master/3569/pipeline-console/)

**Comment by darinpope** [comment]
I'm manually running cd from https://github.com/jenkinsci/bom/network/updates

**Comment by darinpope** [comment]
Changed the cron to see if it has the same success with another full run

**Comment by darinpope** [comment]
[https://ci.jenkins.io/job/Tools/job/bom/job/master/3571/](https://ci.jenkins.io/job/Tools/job/bom/job/master/3571/)

**Comment by MarkEWaite** [comment]
New release: [https://github.com/jenkinsci/bom/releases/tag/3814.v9563d972079a_](https://github.com/jenkinsci/bom/releases/tag/3814.v9563d972079a_)

**Comment by krisstern** [comment]
Thanks @darinpope @gounthar @MarkEWaite for the help in this release!

**Comment by gounthar** [comment]
You're welcome, this is team work!

**Comment by MarkEWaite** [comment]
I learned something with this release of the plugin BOM.  In order to generate a release, the GitHub action requires that at least one of the merged pull requests have a label like `rfe`, `developer`, or `bug`.  Merged pull requests with the `dependencies` label are not enough to generate a release.

**Comment by timja** [comment]
> I learned something with this release of the plugin BOM. In order to generate a release, the GitHub action requires that at least one of the merged pull requests have a label like `rfe`, `developer`, or `bug`. Merged pull requests with the `dependencies` label are not enough to generate a release.

Only the automatic one that gets triggered by the Jenkins build. Manually triggering the workflow in the GitHub UI will avoid that.

Otherwise two options to fix this properly:

1. Add an input to the reusable workflow for `INTERESTING_CATEGORIES` so you can override https://github.com/jenkins-infra/interesting-category-action/blob/78f4b74509528c18790d9c36b2cccb5b21ed3451/action.yaml#L13
2. stop using the reusable workflow for CD and call the actions ourselves

Either is probably fine, cc @jglick

**Comment by MarkEWaite** [comment]
> Only the automatic one that gets triggered by the Jenkins build. Manually triggering the workflow in the GitHub UI will avoid that.

That's another new item for me to have learned.  I manually triggered the release workflow after changing the label of one pull request to `developer`.  I assumed that the label change was the critical thing, but apparently all that I needed to do was manually trigger the release workflow.

> Otherwise two options to fix this properly:
> 
> 1. Add an input to the reusable workflow for `INTERESTING_CATEGORIES` so you can override https://github.com/jenkins-infra/interesting-category-action/blob/78f4b74509528c18790d9c36b2cccb5b21ed3451/action.yaml#L13
> 2. stop using the reusable workflow for CD and call the actions ourselves

What if we changed the label that is assigned by dependabot pull requests so that for this repository they are always labeled as `developer`?  I think that changes in the dependencies of the plugin BOM repository are always interesting changes for plugin BOM releases.

**Comment by timja** [comment]
That would work too

**Comment by MarkEWaite** [comment]
I've submitted a pull request that implements that idea:

* #4105
